clippy_lints/src/utils/conf.rs

```diff
@@ -205,10 +205,41 @@ macro_rules! define_Conf {
 }
 
 define_Conf! {
-    /// Lint: Arithmetic.
+    /// Lint: ARITHMETIC_SIDE_EFFECTS.
     ///
     /// Suppress checking of the passed type names.
+    #[conf_deprecated(
+        "Please use `arithmetic_side_effects_allowed_binary` or `arithmetic_side_effects_allowed_unary` instead",
+        arithmetic_side_effects_allowed_unary
+    )]
     (arithmetic_side_effects_allowed: rustc_data_structures::fx::FxHashSet<String> = <_>::default()),
```

Comment:
I wouldn't say deprecate it for the other two, only if you want to rename it

The description could mention that it's equivalent to specifying both a `_unary` and two `*` `_binary` entries

Reply:
Would also be good to have a test to make sure this option still works

Reply:
Done